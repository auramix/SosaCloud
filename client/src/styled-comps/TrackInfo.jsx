import React from 'react';
import styled from 'styled-components';
import { TrackInteractions } from './TrackInteractions.jsx'
import { TrackButtons } from './TrackButtons.jsx';
import ArtistAnchorEl from './ArtistAnchor.jsx';
import MoreButtonList from './MoreButtonList.jsx';


const TrackInfoDiv = styled.div`
  overflow: hidden;
  font: 14px/1.4 "Lucida Grande","Lucida Sans Unicode","Lucida Sans",Garuda,Verdana,Tahoma,sans-serif;
  font-weight: 100;
  line-height: 1;
  margin: 0 0 2px;
  display: block;
  list-style: none;
  text-decoration: none;
  box-sizing: border-box;
  z-index: 1;
`;


const TrackNameAnchor = styled(TrackInfoDiv)`
  position: relative;
  color: #333;
  margin: 2px 5px 1px 0px;
  line-height: 1.3em;
  font-weight: 50%;
  white-space: nowrap;
`;

const TrackInfo = function (props) {
  let moreDropDown = props.moreClicked && !props.artistPopUp ? <MoreButtonList /> : null;

  return (
    <TrackInfoDiv>
      <TrackButtons moreClickHandler={props.moreClickHandler} moreClicked={props.moreClicked} likeClicked={props.likeClicked} likeClickHandler={props.likeClickHandler} />
      <div>
        <ArtistAnchorEl {...props}/>
      </div>

      <div>
        <TrackNameAnchor as="a" href={""}>
          {props.track.trackName}
        </TrackNameAnchor>
      </div>
      {moreDropDown}

      <TrackInteractions track={props.track} likes={props.likes} />
    </TrackInfoDiv>
  )
}

export { TrackInfo, TrackInfoDiv };