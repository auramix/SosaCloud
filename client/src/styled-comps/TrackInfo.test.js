import { TrackInfo } from './TrackInfo.jsx';

describe('TrackInfo', () => {
  it('Has correct number of children', () => {
    const wrapper = shallow(<TrackInfo track={{ artistName: 'The Dude', trackName: 'Knockin\' on Heaven\'s Door' }} />);

    expect(wrapper.children()).toHaveLength(4);
  });

  // Snapshot test
  it('Has expected snapshot output', () => {
    const wrapper = shallow(
      <TrackInfo track={{ artistName: 'The Dude', trackName: 'Knockin\' on Heaven\'s Door' }} />
    );
    expect(wrapper).toMatchSnapshot();
  })

});