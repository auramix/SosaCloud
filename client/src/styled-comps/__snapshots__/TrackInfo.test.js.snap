// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TrackInfo Has expected snapshot output 1`] = `
<styled.div>
  <TrackButtons />
  <div>
    <ArtistAnchorEl
      track={
        Object {
          "artistName": "The Dude",
          "trackName": "Knockin' on Heaven's Door",
        }
      }
    />
  </div>
  <div>
    <Styled(styled.div)
      as="a"
      href=""
    >
      Knockin' on Heaven's Door
    </Styled(styled.div)>
  </div>
  <TrackInteractions
    track={
      Object {
        "artistName": "The Dude",
        "trackName": "Knockin' on Heaven's Door",
      }
    }
  />
</styled.div>
`;
